src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -78,6 +88,43 @@ final case class AgreementApiServiceImpl(
     }
   }
 
+  private def verifyAttributes(
+    agreement: AgreementManagementDependency.Agreement,
+    eservice: CatalogManagementDependency.EService
+  )(implicit contexts: Seq[(String, String)]): Future[Boolean] = for {
+    consumerAttributes    <- partyManagementService.getPartyAttributes(agreement.consumerId)
+    consumerAttributesIds <- Future.traverse(consumerAttributes)(a =>
+      attributeManagementService.getAttributeByOriginAndCode(a.origin, a.code).map(_.id)
+    )
+    activeEservice        <- CatalogManagementService.validateActivationOnDescriptor(eservice, agreement.descriptorId)
+    result                <- AgreementManagementService.verifyAttributes(
+      consumerAttributesIds,
+      activeEservice.attributes,
+      agreement.verifiedAttributes
+    )
+  } yield result
+
+  private def activate(
+    agreement: AgreementManagementDependency.Agreement,
+    eservice: CatalogManagementDependency.EService,
+    partyId: String
+  )(implicit contexts: Seq[(String, String)]) = for {
+    producer <- partyManagementService.getInstitution(agreement.producerId)
+    consumer <- partyManagementService.getInstitution(agreement.consumerId)
+    document <- pdfCreator.create(agreementTemplate, eservice.name, producer.description, consumer.description)
+    fileInfo = FileInfo("agreementDocument", document.getName, MediaTypes.`application/pdf`)
+    path <- fileManager.store(ApplicationConfiguration.storageContainer, ApplicationConfiguration.agreementDocPath)(
```

Comment:
Nope, the method `storeBytes` should be used, in order to avoid useless file creation